pandas/io/sas/sasreader.py

```diff
@@ -0,0 +1,51 @@
+def read_sas(filepath_or_buffer, format=None, index=None, encoding='utf-8',
+             chunksize=None, iterator=False):
+    """
+    Read SAS files stored as either XPORT or SAS7BDAT format files.
+
+    Parameters
+    ----------
+    filepath_or_buffer : string or file-like object
+        Path to the SAS file.
+    format : string or None
+        If None, file format is inferred.  If 'xport' or 'sas7bdat',
+        uses the corresponding format.
+    index : identifier of index column
+        Identifier of column that should be used as index of the DataFrame.
+    encoding : string
+        Encoding for text data.
+    chunksize : int
+        Read file `chunksize` lines at a time, returns iterator.
+
+    Returns
+    -------
+    DataFrame or SAS7BDATReader or XportReader
+    """
+
+    if format is None:
+        try:
+            fname = filepath_or_buffer.lower()
+            if fname.endswith("xpt"):
+                format = "xport"
+            elif fname.endswith("sas7bdat"):
+                format = "sas7bdat"
+            else:
```

Comment:
`fname.endswith(...)` what if filename is smth. like `d:\data\hello.blah-blahsas7bdat`